File: box/Box/AddIn.cs
using CubeExample.Properties;
using System.ComponentModel;
using System.Runtime.InteropServices;
using Xarial.XCad.Base.Attributes;
using Xarial.XCad.SolidWorks;
using Xarial.XCad.UI.Commands;

namespace CubeExample
{
    [Title("Box Example")]
    public enum Commands_e 
    {
        [Icon(typeof(Resources), nameof(Resources.box_icon))]
        [Title("Create Box")]
        [Description("Creates simple box feature")]
        CreateBox
    }

    [ComVisible(true)]
    [Title("Box Example")]
    [Description("Box example add-in based on xCAD")]
    public class AddIn : SwAddInEx
    {
        public override void OnConnect()
        {
            this.CommandManager.AddCommandGroup<Commands_e>().CommandClick += OnCommandClick;
        }

        private void OnCommandClick(Commands_e spec)
        {
            switch (spec) 
            {
                case Commands_e.CreateBox:
                    this.Application.Documents.Active.Features.CreateCustomFeature<BoxMacroFeatureDef, BoxData, BoxData>();
                    break;
            }
        }
    }
}

File: EventsHandler/vb/AddIn.vb
Imports System.ComponentModel
Imports Xarial.XCad.Documents

<Runtime.InteropServices.ComVisible(True)>
<DisplayName("Event Handlers Example")>
Public Class AddIn
    Inherits Xarial.XCad.SolidWorks.SwAddInEx

    Public Overrides Sub OnConnect()
        Application.Documents.RegisterHandler(Of DocHandler)()
    End Sub
End Class